src/VisualStudio/VisualBasic/Impl/ProjectSystemShim/VisualBasicProject.vb

```diff
@@ -316,7 +317,12 @@ Namespace Microsoft.VisualStudio.LanguageServices.VisualBasic.ProjectSystemShim
             If Not String.IsNullOrEmpty(pCompilerOptions.wszExeName) Then
                 VisualStudioProject.AssemblyName = Path.GetFileNameWithoutExtension(pCompilerOptions.wszExeName)
                 If Not String.IsNullOrEmpty(pCompilerOptions.wszOutputPath) Then
-                    VisualStudioProject.IntermediateOutputFilePath = Path.Combine(pCompilerOptions.wszOutputPath, pCompilerOptions.wszExeName)
+                    Dim outputFilePath = Path.Combine(pCompilerOptions.wszOutputPath, pCompilerOptions.wszExeName)
+                    VisualStudioProject.IntermediateOutputFilePath = outputFilePath
+
+                    ' The Project System doesn't indicate whether we emit PDB, what kind of PDB we emit nor the path of the PDB.
+                    ' To work around we look for the PDB on the path specified in the PDB debug directory.
+                    CType(Workspace, VisualStudioWorkspaceImpl).SetCompilationOutputs(VisualStudioProject.Id, New VisualStudioCompilationOutputFiles(outputFilePath))
```

Comment:
Won't this get blown away if something else causes the options processor to overwrite it?

Reply:
My rather limited understanding is that VB legacy project does not use command line to set options. Is that incorrect?

Reply:
I mean for setting `/out` option - this is set by this method, no? The options processor does not set the compilation outputs if `/out` is not specified on the command line.

Reply:
I see. `_project.IntermediateOutputFilePath` will be non-null in that case, so the processor will overwrite this. Hmm, I think I'm just gonna use `VisualStudioCompilationOutputFiles` everywhere.